config/initializers/locale.rb

```diff
@@ -0,0 +1,6 @@
+=begin
+I18n.load_path += Dir[Rails.root.join('config','locales','**','*.{rb,yml}').to_s]
+
+I18n.available_locales=[:en,:ja]
+I18n.default_locale = :ja
+=end
```

Comment:
使わないファイルなら消しましょう。何か残したい理由があるんでしょうか?

Reply:
本ファイル自体は結局不要だったので削除でします。

ただし

ガイドでhttps://railsguides.jp/i18n.html#i18n%E3%83%A2%E3%82%B8%E3%83%A5%E3%83%BC%E3%83%AB%E3%82%92%E8%A8%AD%E5%AE%9A%E3%81%99%E3%82%8B

config/application.rbでは次のように、デフォルトのロケールを変更したり訳文読み込みパスを設定したりできます。
〜
①読み込みパスは、訳文の参照前に指定する必要があります。config/application.rbのイニシャライザのデフォルトロケールを変更するには次のようにします。

``` rb
# config/initializers/locale.rbファイルの内容:

# I18nライブラリに訳文の探索場所を指示する
I18n.load_path += Dir[Rails.root.join('lib', 'locale', '*.{rb,yml}')]

# アプリケーションでの利用を許可するロケールをホワイトリスト化する
I18n.available_locales = [:en, :pt]

# ロケールを:en以外に変更する
I18n.default_locale = :pt
```
②ただし、アプリケーションのi18nで設定する代わりにI18n.load_pathsに直接追加すると、外部gem由来の訳文はオーバーライドされません。

とかいてありますが、①、②はどういう意味でしょうか？

Reply:
わかりにくい表現ですねー。僕も読んでもよくわかりません。ただ、原文だとこうなってます。

①読み込みパスは、訳文の参照前に指定する必要があります。config/application.rbのイニシャライザのデフォルトロケールを変更するには次のようにします。
The load path must be specified before any translations are looked up. To change the default locale from an initializer instead of config/application.rb:

instead of config/application.rb なので、単にconfig/application.rbに書く設定はinitializerに書いてもいいよって言ってるだけかと。

②ただし、アプリケーションのi18nで設定する代わりにI18n.load_pathsに直接追加すると、外部gem由来の訳文はオーバーライドされません。
Note that appending directly to I18n.load_path instead of to the application's configured i18n will not override translations from external gems.

原文読んでも難しいですが、`will not override translations from external gems.`ですし、I18n.load_pathがどうあれ、外部gemの訳文は上書きしないよって言ってるのかなーと。

いずれにせよ日本語訳はどっちもちょっと変だと思います。

https://guides.rubyonrails.org/i18n.html#configure-the-i18n-module

Reply:
Railsのドキュメントの日本語は時折怪しいので、なんだかわからんなーってときは今回のように原文あたってみたほうがいいかもです。